plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegateTest.java

```diff
@@ -294,18 +301,48 @@ public void testGenerateRunConfiguration_withProgramArgs() throws CoreException
         .getAttribute(eq(IJavaLaunchConfigurationConstants.ATTR_PROGRAM_ARGUMENTS), anyString());
   }
 
+  @Test
+  public void testGenerateRunConfiguration_withEnvironment() throws CoreException {
+    Map<String, String> definedMap = new HashMap<>();
+    definedMap.put("foo", "bar");
+    definedMap.put("baz", "${env_var:PATH}");
+    when(launchConfiguration.getAttribute(eq(ILaunchManager.ATTR_ENVIRONMENT_VARIABLES),
+        anyMapOf(String.class, String.class)))
+        .thenReturn(definedMap);
+
+    DefaultRunConfiguration config = new LocalAppEngineServerLaunchConfigurationDelegate()
+        .generateServerRunConfiguration(launchConfiguration, server, ILaunchManager.RUN_MODE);
+
+    Map<String, String> parsedEnvironment = config.getEnvironment();
+    assertNotNull(parsedEnvironment);
+    assertEquals("bar", parsedEnvironment.get("foo"));
+    assertEquals(System.getenv("PATH"), parsedEnvironment.get("baz"));
+    verify(launchConfiguration).getAttribute(eq(ILaunchManager.ATTR_ENVIRONMENT_VARIABLES),
+        anyMapOf(String.class, String.class));
+    verify(launchConfiguration, atLeastOnce())
```

Comment:
I don't see much value in these `verify` checks. We already check the values of `parsedEnvironment`.

Reply:
We've been using them pretty systematically throughout.